pennylane/interfaces/torch.py

```diff
@@ -246,3 +258,289 @@ def execute(tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_d
         max_diff=max_diff,
     )
     return ExecuteTapes.apply(kwargs, *parameters)
+
+
+def pytreeify(cls):
+    """Pytree class"""
+    orig_fw = cls.forward
+    orig_bw = cls.backward
+    orig_apply = cls.apply
+
+    def new_apply(*inp):
+        # Inputs already flat
+        out_struct_holder = []
+        flat_out = orig_apply(out_struct_holder, *inp)
+        return pytree.tree_unflatten(flat_out, out_struct_holder[0])
+
+    def new_forward(ctx, out_struct_holder, *inp):
+        out = orig_fw(ctx, *inp)
+        flat_out, out_struct = pytree.tree_flatten(out)
+        ctx._out_struct = out_struct
+        out_struct_holder.append(out_struct)
+        return tuple(flat_out)
+
+    def new_backward(ctx, *flat_grad_outputs):
+        grad_outputs = pytree.tree_unflatten(flat_grad_outputs, ctx._out_struct)
+        grad_inputs = orig_bw(ctx, *grad_outputs)
+        return (None,) + tuple(grad_inputs)
+
+    cls.apply = new_apply
+    cls.forward = new_forward
+    cls.backward = new_backward
+    return cls
+
+
+def _compute_vjps_new(dys, jacs, multi_measurements, device=None):
+    """Compute the vjps of multiple tapes, directly for a Jacobian and tangents."""
+    # pylint: disable=unused-argument
+    # TODO: add device
+    vjps = []

```

Comment:
Is this meant to be added later on?

Reply:
Removed it will be changed for shot vectors.